src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/AssemblyResolver.cs

```diff
@@ -0,0 +1,99 @@
+    using System.Runtime.Loader;
+
+    public class AssemblyResolver : IAssemblyResolver
+    {
+        private AssemblyResolverUtility resolverUtility;
+
+        /// <summary>
+        /// Specifies whether the resolver is disposed or not
+        /// </summary>
+        private bool isDisposed;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AssemblyResolver"/> class.
+        /// </summary>
+        /// <param name="directories"> The search directories. </param>
+        [System.Security.SecurityCritical]
+        public AssemblyResolver(IEnumerable<string> directories)
+        {
+            this.resolverUtility = new AssemblyResolverUtility(directories);
+
+            AssemblyLoadContext.Default.Resolving += this.AssemblyResolverEvent;
+        }
+
+        /// <inheritdoc />
+        ~AssemblyResolver()
+        {
+            this.Dispose(false);
+        }
+
+        /// <summary>
+        /// Set the directories from which assemblies should be searched
+        /// </summary>
+        /// <param name="directories"> The search directories. </param>
+        [System.Security.SecurityCritical]
+        public void AddSearchDirectories(IEnumerable<string> directories)
+        {
+            this.resolverUtility.AddSearchDirectories(directories);
+        }
+
+        /// <summary>
+        /// Assembly Resolve event handler for App Domain - called when CLR loader cannot resolve assembly.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private Assembly AssemblyResolverEvent(object sender, object eventArgs)
+        {
+            AssemblyName args = eventArgs as AssemblyName;
+
+            return args == null ? null : this.resolverUtility.AssemblyResolverEventHandler(args.Name);
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+
+            // Use SupressFinalize in case a subclass
+            // of this type implements a finalizer.
+            GC.SuppressFinalize(this);
+        }
+
+        [System.Security.SecurityCritical]
+        protected void Dispose(bool disposing)
+        {
+            if (!this.isDisposed)
+            {
+                if (disposing)
+                {
+                    AssemblyLoadContext.Default.Resolving -= this.AssemblyResolverEvent;
+                }
+
+                this.isDisposed = true;
+            }
+        }
+
+        public void SetCurrentDomainAssemblyResolve()
+        {
+            AssemblyLoadContext.Default.Resolving += this.CurrentDomainAssemblyResolve;
+        }
+
+        private Assembly CurrentDomainAssemblyResolve(AssemblyLoadContext loadContext, AssemblyName args)
```

Comment:
Can we move all the common code of `netcore/System/AssemblyResolver.cs` and `net46/System/AssemblyResolver.cs ` in `common\System.AssemblyResolver.cs`